compiler/rustc_error_codes/src/error_codes/E0311.md

```diff
@@ -0,0 +1,52 @@
+This error occurs when there is an unsatisfied outlives bound involving an
+elided region on a generic type parameter or associated type.
+
+Erroneous code example:
+
+```compile_fail,E0311
+fn no_restriction<T>(x: &()) -> &() {
+    with_restriction::<T>(x)
+}
+
+fn with_restriction<'a, T: 'a>(x: &'a ()) -> &'a () {
+    x
+}
+```
+
+Why doesn't this code compile? It helps to look at the lifetime bounds that are
+automatically adding by the compiler. For more details see the Rust
+Documentation for Lifetime Elision:
+https://doc.rust-lang.org/reference/lifetime-elision.html.
+
+There are two lifetimes being passed into the `no_restriction()` function: one
+associated with the generic type `T` parameter and the other with the input
+argument `x`. The compiler does not know which of these lifetimes can be
+assigned to the output reference, so we get an error.
```

Comment:
Let me know if there is a more clear or correct way to explain this concept.

Reply:
I replaced my description with yours and then applied a few minor changes to smooth out the wording a bit. 

For the reference link, the best I could find is for the RFD on implied bounds. Let me know if you have a better idea.
https://rust-lang.github.io/rfcs/2089-implied-bounds.html